client/src/featureform/serving.py

```diff
@@ -142,7 +142,20 @@ def features(self, features, entities, model: Union[str, Model] = None):
         if model is not None:
             req.model.name = model if isinstance(model, str) else model.name
         resp = self._stub.FeatureServe(req)
-        return [parse_proto_value(val) for val in resp.values]
+
+        feature_values = []
+        for val in resp.values:
+            parsed_value = parse_proto_value(val)
+            
+            is_ondemand_feature = type(parsed_value) == bytes
```

Comment:
Is there a different way to check the type? We may want to return bytes as a value in the future

Reply:
the alternative will be for each val, we will have to pull the metadata for that feature in that position and see the category if is on demand_feature. 

```
for i, val in enumerate(resp.values):
      feature = features[i]
      category = self._stub.PullFeatureMetadata(feature) # need to figure out right method for this
      
      if category == "ONDEMAND_FEATURE":
```

another approach is we modify FeatureServe to return values and categories.

Reply:
Isnt there a way to check what the oneof type is?